core/controlplane/config/config.go

```diff
@@ -435,6 +438,7 @@ type DeploymentSettings struct {
 	CloudWatchLogging       `yaml:"cloudWatchLogging,omitempty"`
 	AmazonSsmAgent          `yaml:"amazonSsmAgent,omitempty"`
 	CloudFormationStreaming bool `yaml:"cloudFormationStreaming,omitempty"`
+	KubeDnsMasq             `yaml:"kubeDnsMasq,omitempty"`
```

Comment:
How about renaming things so that the whole expression look like `NodeLocalKubeDnsMasq.Enabled` rather than `KubeDnsMasq.DaemonSet`?
I guess it will then better/more directly represents the fact that a dns masq pod is created for and per each node.